packages/components/atoms/f-button/stories/IconButton.stories.js

```diff
@@ -1,88 +1,127 @@
+    <div>
+        <div class="u-spacingBottom--large storybook-grid storybook-grid-columns--4 storybook-grid-stack--lessThanWide"
+        v-for="(list, index) in iconButtonLists">
+            <f-button
+                class="u-spacingBottom--large"
+                v-for="(icon, index) in list"
+                :key="index"
+                :buttonType="icon.type"
+                :buttonSize="icon.size"
+                :actionType="actionType"
+                :href="href"
+                :isIcon="true"
+                :isLoading="isLoading">
+                <CrossIcon />
+                <span class="is-visuallyHidden">
+                    {{ icon.text }}
+                </span>
+            </f-button>
 
-        <f-button
-            class="u-spacingRight u-spacingBottom--large"
-            :buttonType="buttonType"
-            :buttonSize="buttonSize"
-            :actionType="actionType"
-            :href="href"
-            :isIcon="true"
-            :isLoading="isLoading">
-            <PlusIcon />
-        </f-button>
+            <f-button
+                class="u-spacingBottom--large"
+                v-for="(icon, index) in list"
+                :key="index"
+                :buttonType="icon.type"
+                :buttonSize="icon.size"
+                :actionType="actionType"
+                :href="href"
+                :isIcon="true"
+                :isLoading="isLoading">
+                <PlusIcon />
+                <span class="is-visuallyHidden">
+                    {{ icon.text }}
+                </span>
+            </f-button>
 
-        <f-button
-            class="u-spacingRight u-spacingBottom--large"
-            :buttonType="buttonType"
-            :buttonSize="buttonSize"
-            :actionType="actionType"
-            :href="href"
-            :isIcon="true"
-            :isLoading="isLoading">
-            <MoreVerticalIcon />
-        </f-button>
+            <f-button
+                class="u-spacingBottom--large"
+                v-for="(icon, index) in list"
+                :key="index"
+                :buttonType="icon.type"
+                :buttonSize="icon.size"
+                :actionType="actionType"
+                :href="href"
+                :isIcon="true"
+                :isLoading="isLoading">
+                <MoreVerticalIcon />
+                <span class="is-visuallyHidden">
+                    {{ icon.text }}
+                </span>
+            </f-button>
+        </div>
     </div>`
 });
+IconButtonComponent.argTypes = {
+    isLoading: SharedButtonArgTypes.isLoading,
+    href: SharedButtonArgTypes.href,
+    actionType: {
+        control: { type: 'select', options: ['button', 'submit', 'reset'] },
+        description: 'Choose the action type of the button',
+        defaultValue: 'button'
+    }
+};
+
+IconButtonComponent.args = {
+    iconButtonLists: [
+        [
+            { type: 'primary', size: 'large', text: 'Large Primary Icon Button' },
```

Comment:
LGTM for now, but I'm wondering if it might be better in future to have this structure built programmatically, like nested "for each type in types", "for each size in sizes", etc, then if you want to add a new size you can just update the sizes array rather than adding a new object for each type.

Reply:
ah yep agree that would definitely be better and more future proof